### PR TITLE
benchmark each module in their own node process

### LINKED ISSUE #864: Order of benchmarks seems to bias results
Hi, thanks for this project.

I'm currently looking at submitting a new runtime type checker for comparative performance benchmarks, but have noticed that the order in which each validator is run seems to degrade subsequent benchmark results. By moving performing tests around, I'm able  to greatly improve the performance results of tests by simply running them first.

I'm thinking this is likely due to some of the validators utilizing more memory overhead (implicating the GC), or that V8 optimization heuristics have broken down causing V8 to take the slow path (this is quite likely given how validation routines may require dynamic property access on objects to complete). To rule out the GC, I started the project with `npm run compile:spectypes && node --expose-gc -r ts-node/register ./index.ts` and called `global.gc()` to force collection prior to tests running, but no luck. So the problem is most likely related V8 optimization heuristics. 

I think to give all tests an even footing, I'm wondering if it would be possible to reshape these tests such that each is run within it's own node process (rather than running all tests in the same process). This would give each a fresh instance in which to do it's thing, and should yield more accurate results.

Open to thoughts if this is something worth looking into.

### PR DESCRIPTION
**Pull request by hoeck**
trying to fix #864

### CODE REVIEW

File: index.ts [hunk 1]
```diff
@@ -1,4 +1,69 @@
...
+          await cases.importCase(c);
+        }
+
+        await benchmarks.runAllBenchmarks();
+      }
+      break;
+
+    default:
+      console.error('unknown command:', command);
+      process.exit(1);
```
Comment (moltar) [inline index.ts:63]: > Don't use process.exit(); throw an error instead

eslint-ignore it?

File: test/benchmarks.test.ts [hunk 2]
```diff
@@ -1,5 +1,47 @@
...
+import '../cases/ts-utils';
+import '../cases/typeofweb-schema';
+import '../cases/valita';
+import '../cases/yup';
+import '../cases/zod';
+
+test('all cases must have been imported in tests', () => {
+  const registeredCases = new Set<string>();
+
+  getRegisteredBenchmarks().forEach(([benchmarkId, benchmarkCases]) => {
```
Comment (moltar) [inline test/benchmarks.test.ts:37]: > 'benchmarkId' is defined but never used

**Review by moltar — CHANGES_REQUESTED** [review]
Some lint issues but otherwise great work!

**Comment by tonivj5** [comment]
Very excited to see the "real" ones result 👀👏🏻

**Comment by moltar** [comment]
Oops, sorry, I've introduced a small conflict by merging #865

**Comment by hoeck** [comment]
@moltar fixed all linter issues (atm. I am not using Eslint at work so I always forget about it on other projects :sweat_smile:)

Rebased, added a message linking to this issue and disabled an Eslint rule that was complaining about the way spectypes are imported.

**Comment by hoeck** [comment]
Sorry for the noise, I should have used a Github markdown previewer :expressionless: :grinning:.

**Review by moltar — APPROVED** [review]

**Comment by moltar** [comment]
Sweet @hoeck! 

> I am not using Eslint at work 

Do you use something else?

**Comment by sinclairzx81** [comment]
@hoeck Good work :) +1

**Comment by hoeck** [comment]
> Do you use something else?

No, we're only using prettier in our team of two. There have been always more important tasks than fixing the old and ignored tslint setup :smile: .

**Pull request merged**